File: trees/bst_to_greater.cpp
void solve(TreeNode* &root,int& mx){
    if(root == NULL) return;
    solve(root->right,mx);
    root->val = root->val + mx;
    mx = root->val;
    solve(root->left,mx);
}

TreeNode* convertBST(TreeNode *root){
    int mx = 0;
    solve(root,mx);
    return root;
}

File: trees/bst_to_greater.py
def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        tot=0
        def dfs(node):
            nonlocal tot
            if not node:
                return 
            if node.right:
                dfs(node.right)
            tot=node.val=tot+node.val
            if node.left:
                dfs(node.left)
            return
        dfs(root)
        return root
